src/components/WebCLI/connection.ts

```diff
@@ -115,28 +112,12 @@ class Connection {
       if (!Array.isArray(data.output)) {
         throw new Error(Label.INCORRECT_DATA_ERROR);
       }
-      this.#pushToMessageBuffer(`\n${data.output[0]}`);
+      // Build up messages until it gets the 'done' message:
+      this.#messageBuffer.push(data.output[0]);
```

Comment:
Send the content all at once instead of throttling the content and incrementally returning it.